step-12 compilation fails

By bisecting I believe the commit that caused this issue is bb57e1883b.

ping @Rombur I think MeshWorker defines FilteredIterator to mean something else but I am not sure of this.

For reference, here is the error:
```
In file included from /home/bangerth/p/deal.II/1/install/examples/step-12/step-12.cc:69:0:
/home/bangerth/p/deal.II/1/install/include/deal.II/meshworker/loop.h:55:40: error: ‘FilteredIterator’ does not name a type
   inline bool is_active_iterator(const FilteredIterator<TriaActiveIterator<ACCESSOR> > &)
                                        ^
/home/bangerth/p/deal.II/1/install/include/deal.II/meshworker/loop.h:55:40: error: ISO C++ forbids declaration of ‘parameter’ with no type [-fpermissive]
/home/bangerth/p/deal.II/1/install/include/deal.II/meshworker/loop.h:55:56: error: expected ‘,’ or ‘...’ before ‘<’ token
   inline bool is_active_iterator(const FilteredIterator<TriaActiveIterator<ACCESSOR> > &)
                                                        ^
make[2]: *** [CMakeFiles/step-12.dir/step-12.cc.o] Error 1
make[1]: *** [CMakeFiles/step-12.dir/all] Error 2
make: *** [all] Error 2

```

--- a/include/deal.II/meshworker/loop.h
+++ b/include/deal.II/meshworker/loop.h
@@ -1,6 +1,6 @@
 // ---------------------------------------------------------------------
 //
-// Copyright (C) 2006 - 2015 by the deal.II authors
+// Copyright (C) 2006 - 2016 by the deal.II authors
 //
 // This file is part of the deal.II library.
 //
@@ -22,6 +22,7 @@
 #include <deal.II/base/work_stream.h>
 #include <deal.II/base/template_constraints.h>
 #include <deal.II/grid/tria.h>
+#include <deal.II/grid/filtered_iterator.h>
 #include <deal.II/meshworker/local_integrator.h>
 #include <deal.II/meshworker/dof_info.h>
 #include <deal.II/meshworker/integration_info.h>
@@ -32,7 +33,6 @@
 DEAL_II_NAMESPACE_OPEN
 
 template <typename> class TriaActiveIterator;
-template <typename> class FilteredIterator;
 
 namespace internal
 {
@@ -52,7 +52,7 @@ namespace internal
   }
 
   template <class ACCESSOR>
-  inline bool is_active_iterator(const FilteredIterator<TriaActiveIterator<ACCESSOR> > &)
+  inline bool is_active_iterator(const dealii::FilteredIterator<TriaActiveIterator<ACCESSOR> > &)
   {
     return true;
   }

